monkey/monkey_island/cc/app.py

```diff
@@ -111,7 +113,17 @@ def init_app_url_rules(app):
     app.add_url_rule("/<path:static_path>", "serve_static_file", serve_static_file)
 
 
-def init_api_resources(api):
+class FlaskDIWrapper:
+    def __init__(self, api: flask_restful.Api, container: DIContainer):
+        self._api = api
+        self._container = container
+
+    def add_resource(self, resource: Type[flask_restful.Resource], *urls: str):
```

Comment:
So this allows to dynamically resolve services for resources, but how would we dynamically add resources?

Reply:
The infrastructure for dynamically adding resources is TBD, but, generally speaking, the user would provide a python file that contains the resource and a manifest file (probably YAML) that gives the system some metadata about the resource. These files are placed in a certain location and the island scans that directory and loads the custom resource when it first starts up. Again, specifics are TBD.